cluster-autoscaler/utils/gpu/gpu.go

```diff
@@ -68,6 +68,24 @@ func FilterOutNodesWithUnreadyGpus(allNodes, readyNodes []*apiv1.Node) ([]*apiv1
 	return newAllNodes, newReadyNodes
 }
 
+// GetGpuTypeForMetrics returns name of the GPU used on the node or empty string if there's no GPU
+// if the GPU type is unknown, "generic" is returned
+// NOTE: current implementation is GKE/GCE-specific
+func GetGpuTypeForMetrics(node *apiv1.Node) string {
+	// we use the GKE label if there is one
+	gpuType, found := node.Labels[GPULabel]
```

Comment:
IIUC the issue here is that just because a node has this label doesn't mean it'll have GPU. We treat such nodes as not having a GPU when making scale-up/scale-down decisions. Our metrics should hopefully be consistent with it (or alternatively, report it as a possible misconfiguration.)

Reply:
I agree scale-up and scale-down should be consistent as much as possible. What would be the inconsistency if we added checking template here?

Reply:
Please have a look at PR #1017 which addresses this discussion.